filebeat/input/journald/config.go

```diff
@@ -48,6 +48,10 @@ type config struct {
 	// MaxBackoff is the limit of the backoff time.
 	MaxBackoff time.Duration `config:"max_backoff" validate:"min=0,nonzero"`
 
+	// Since is the relative time offset from now to provide journal
+	// entries from. If Since is nil, no offset is applied.
+	Since *time.Duration `config:"since"`
```

Comment:
This is not correct. A zero since is a valid offset, perhaps not a very useful one, but it is valid and means start reading the logs from the entry now.

Reply:
Ok, please let me help to understand how the configuration would work:
1. If `since` is `nil`, then `cursor` and `cursor_seek_fallback` need to set in a way that `since` will not be used.
2. If `since` is not `nil`, then `cursor_seek_fallback` need to set in a way that `since` will always be used, which will always enforce that https://github.com/elastic/beats/blob/b673122ee89ca30ae8e11b362d06b3d7b71a1e9a/filebeat/input/journald/input.go#L147 will not panic  when dereferencing `inp.Since`.

Is that it?

Reply:
Yes, that's correct. We cannot be in a code path where `*inp.Since` would be nil because we must have run the config validation gauntlet.